src/workloads/query/QuerySettings.yml

```diff
@@ -0,0 +1,164 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  Measure the performance for find command with-/out query settings as well as checking both hit and miss cases.
+
+KeyWords:
+- querySettings
+- find
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 400
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection querySettingsColl
+  DocumentCount: &DocumentCount 100000
+  MaxPhases: &MaxPhases 12
+  HitFilter: &HitFilter {a1: {^RandomInt: {min: 1, max: 350}}}
+  MissFilter: &MissFilter {a2: {^RandomInt: {min: 1, max: 350}}}
+  Threads: &Threads 64
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a1: &integer {^RandomInt: {min: -100, max: 100}}
+          a2: *integer
+        Indexes:
+        - keys: {a1: 1}
+        - keys: {a2: 1}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2, 4, 6, 8, 10, 12]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: FindWithoutQuerySettings
+  Type: CrudActor
+  Threads: *Threads
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: findOne
+          OperationCommand:
+            Filter: *HitFilter
```

Comment:
Even if the pseudo-random number generator seed is fixed for Genny test run, it is not clear if the filter matches a document. That likely can become very important if we want to compare results between no query settings, matching and non-matching query settings due to low resource consumption of the "find" command. I find this randomness and uncertainty unnecessary in this already messy world :).
I would suggest explicitly testing a scenario in which at least one document matches.

Reply:
That's correct.